core/src/test/java/greencity/controller/ShoppingListItemControllerTest.java

```diff
@@ -0,0 +1,177 @@
+package greencity.controller;
+
+import greencity.converters.UserArgumentResolver;
+import greencity.dto.shoppinglistitem.ShoppingListItemDto;
+import greencity.dto.user.UserShoppingListItemResponseDto;
+import greencity.dto.user.UserVO;
+import greencity.service.ShoppingListItemService;
+import greencity.service.UserService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.modelmapper.ModelMapper;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+import org.springframework.validation.Validator;
+
+import java.security.Principal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+import static greencity.ModelUtils.getPrincipal;
+import static greencity.ModelUtils.getUserVO;
+import static org.mockito.Mockito.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@ExtendWith(MockitoExtension.class)
+public class ShoppingListItemControllerTest {
+
+    private MockMvc mockMvc;
+
+    @Mock
+    private ShoppingListItemService shoppingListItemService;
+
+    @Mock
+    private Validator mockValidator;
+
+    @Mock
+    private UserService userService;
+
+    @Mock
+    private ModelMapper modelMapper;
+
+    @InjectMocks
+    private ShoppingListItemController shoppingListItemController;
+    private static final String initialUrl = "/user/shopping-list-items";
+    private final Principal principal = getPrincipal();
+    private final UserVO userVO = getUserVO();
+
+    @BeforeEach
+    void setup() {
+        this.mockMvc = MockMvcBuilders.standaloneSetup(shoppingListItemController)
+                .setValidator(mockValidator)
+                .setCustomArgumentResolvers(new UserArgumentResolver(userService, modelMapper))
+                .build();
+    }
+
+    @Test
+    void bulkDeleteUserShoppingListItem_StatusOk() throws Exception {
+
```

Comment:
it is not a correct method. According to name, it is supposed to test bulkDeleteUserShoppingListItem but in this call "mockMvc.perform(delete(initialUrl)" you ask for the method with the name "delete" from the controller. The reason is you use "initialUrl" but it should be "initialUrl + /user-shopping-list-items" and also "bulkDeleteUserShoppingListItems" requires other params in the request.

Reply:
Thanks, fixes are done